stable_baselines3/dqn/policies.py

```diff
@@ -0,0 +1,267 @@
+from typing import Optional, List, Callable, Union, Type, Any, Dict
+
+import gym
+import numpy as np
+import torch as th
+import torch.nn as nn
+from stable_baselines3.common.policies import (BasePolicy, register_policy, create_mlp,
+                                               NatureCNN, BaseFeaturesExtractor, FlattenExtractor)
+
+
+class QNetwork(BasePolicy):
+    """
+    Action-Value (Q-Value) network for DQN
+
+    :param observation_space: (gym.spaces.Space) Observation space
+    :param action_space: (gym.spaces.Space) Action space
+    :param net_arch: (Optional[List[int]]) The specification of the policy and value networks.
+    :param device: (str or th.device) Device on which the code should run.
+    :param activation_fn: (Type[nn.Module]) Activation function
+    :param epsilon: (float) Exploration fraction (for the epsilon-greedy exploration)
+    :param normalize_images: (bool) Whether to normalize images or not,
+         dividing by 255.0 (True by default)
+    """
+
+    def __init__(self, observation_space: gym.spaces.Space,
+                 action_space: gym.spaces.Space,
+                 features_extractor: nn.Module,
+                 features_dim: int,
+                 net_arch: Optional[List[int]] = None,
+                 device: Union[th.device, str] = 'auto',
+                 activation_fn: Type[nn.Module] = nn.ReLU,
+                 epsilon: float = 0.05,
+                 normalize_images: bool = True):
+        super(QNetwork, self).__init__(observation_space, action_space,
+                                       features_extractor=features_extractor,
+                                       normalize_images=normalize_images,
+                                       device=device)
+
+        if net_arch is None:
```

Comment:
This defaulting would cause headache (e.g. you give CNN policy, and now you suddenly have network with layers NatureCNN - 64 - 64 - output, while users expect NatureCNN - output). This seems to be a common trend around rest of the code that I think should be fixed.

Reply:
it won't be the case normally (see line 135), what do you propose instead?

Reply:
Oh man, a proper brainfart. For some reason I have missed this in all of the code I have been going over. It looks good as it is, my bad!